test/integration/network_reserved_ips_test.go

```diff
@@ -2,150 +2,199 @@ package integration
+	client, teardown := createTestClient(t, "fixtures/TestReservedIPAddresses_EndToEndTest")
 	defer teardown()
 
 	filter := ""
-	ipList, listIpErr := client.GetReservedIPs(context.Background(), NewListOptions(0, filter))
-	if listIpErr != nil {
-		t.Logf("Error listing ipaddresses, expected struct, got error %v", listIpErr)
-	} else {
-		if len(ipList) == 0 {
-			t.Logf("The customer has not reserved an IP %v", ipList)
-		}
+
+	ipList, err := client.ListReservedIPAddresses(context.Background(), NewListOptions(0, filter))
+
+	if err != nil {
+		t.Fatalf("Error listing IP addresses: %v", err)
 	}
 
+	initialCount := len(ipList)
+
 	// Attempt to reserve an IP
-	reserveIP, reserveIpErr := client.ReserveIPAddress(context.Background(), ReserveIPOptions{
+	resIP, resErr := client.ReserveIPAddress(context.Background(), ReserveIPOptions{
 		Region: "us-east",
 	})
-	if reserveIpErr != nil {
-		t.Logf("Failed to reserve IP: %v", reserveIpErr)
-	} else {
-		t.Logf("Successfully reserved IP: %+v", reserveIP)
+
+	if resErr != nil {
+		t.Fatalf("Failed to reserve IP. This test expects the user to have 0 prior reservations and the ip_reservation_limit to be 2. Error from the API: %v", resErr)
 	}
 
-	if reserveIP != nil {
-		// Fetch the reserved IP
-		reservedIP, fetchIpErr := client.GetReservedIPAddress(context.Background(), reserveIP.Address)
-		if fetchIpErr != nil {
-			t.Logf("Error getting ipaddress, expected struct, got %v and error %v", reservedIP, fetchIpErr)
-		}
+	if resIP == nil {
+		t.Fatalf("Reserved IP is nil")
+	}
 
-		// Verify the list of IPs has increased
-		verifyList, verifyListIpErr := client.GetReservedIPs(context.Background(), NewListOptions(0, filter))
-		if verifyListIpErr != nil {
-			t.Logf("Error listing ipaddresses, expected struct, got error %v", verifyListIpErr)
-		} else {
-			if len(verifyList)-len(ipList) == 1 {
-				t.Log("Increase in IP list confirmed", verifyList)
-			} else {
-				t.Errorf("Increase in IP list not confirmed")
-			}
-		}
+	t.Logf("Successfully reserved IP: %+v", resIP)
 
-		// Delete the reserved IP
-		if reserveIP != nil {
-			deleteErr := client.DeleteReservedIPAddress(context.Background(), reserveIP.Address)
-			if deleteErr != nil {
-				t.Logf("Error deleting reserved IP address: %v", deleteErr)
-			} else {
-				t.Logf("Successfully deleted reserved IP address: %s", reserveIP.Address)
-			}
-		}
+	// Fetch the reserved IP
+	fetchedIP, fetchErr := client.GetReservedIPAddress(context.Background(), resIP.Address)
+	if fetchErr != nil {
+		t.Errorf("Error getting reserved IP address: %v", fetchErr)
+	}
 
-		// Verify the IP has been deleted
-		if reserveIP != nil {
-			verifyDeletedIP, fetchDeletedIpErr := client.GetReservedIPAddress(context.Background(), reserveIP.Address)
-			if fetchDeletedIpErr != nil {
-				t.Logf("Error getting ipaddress, expected struct, got %v and error %v", verifyDeletedIP, fetchDeletedIpErr)
-			}
+	if fetchedIP == nil {
+		t.Errorf("Retrieved reserved IP is nil")
```

Comment:
It's helpful to always include the expected state in the error messages since it simplifies debugging in the future.

Reply:
I have now included the IP Address which the test failed to retrieve in the error message.